### PR TITLE
feat: added an option to show bootstrap server host of the Kafka instance

### LINKED ISSUE #636: Add option to show the bootstrapServer
### Feature or problem description

Add an option for the command to return the bootstrapServer hostname of the instance in use

### Proposed solution

rhoas kafka cluster describe --bootstrapserver

### Additional context

When getting the information through the CLI it can be easy to just return the value without the need to parse the json returned.

### PR DESCRIPTION
**Pull request by alexal**
Closes #636

### CODE REVIEW

File: pkg/cmd/kafka/describe/describe.go [hunk 1]
```diff
@@ -88,6 +89,7 @@ func NewDescribeCommand(f *factory.Factory) *cobra.Command {
 	cmd.Flags().StringVarP(&opts.outputFormat, "output", "o", "json", opts.localizer.MustLocalize("kafka.common.flag.output.description"))
 	cmd.Flags().StringVar(&opts.id, "id", "", opts.localizer.MustLocalize("kafka.describe.flag.id"))
 	cmd.Flags().StringVar(&opts.name, "name", "", opts.localizer.MustLocalize("kafka.describe.flag.name"))
+	cmd.Flags().BoolVar(&opts.bootstrapServer, "bootstrapserver", false, opts.localizer.MustLocalize("kafka.describe.flag.bootstrapserver"))
```
Comment (craicoverflow) [inline pkg/cmd/kafka/describe/describe.go:92]:
```suggestion
	cmd.Flags().BoolVar(&opts.bootstrapServer, "bootstrap-server", false, opts.localizer.MustLocalize("kafka.describe.flag.bootstrapserver"))
```

File: pkg/cmd/kafka/describe/describe.go [hunk 2]
```diff
@@ -125,7 +127,11 @@ func runDescribe(opts *Options) error {
 		}
 	}
 
-	return printKafka(kafkaInstance, opts)
+	if !opts.bootstrapServer {
+		return printKafka(kafkaInstance, opts)
+	}
+	opts.Logger.Info(*kafkaInstance.BootstrapServerHost)
```
Comment (craicoverflow) [inline pkg/cmd/kafka/describe/describe.go:133]: It needs to print to stdout.

The logger only prints to stderr, which is not machine-readable and so is not useful for this.

File: pkg/cmd/kafka/describe/describe.go [hunk 3]
```diff
@@ -125,7 +127,11 @@ func runDescribe(opts *Options) error {
 		}
 	}
 
-	return printKafka(kafkaInstance, opts)
+	if !opts.bootstrapServer {
```
Comment (craicoverflow) [inline pkg/cmd/kafka/describe/describe.go:130]: I'd rather see `if opts.bootstrapServer` and print the url and return, leaving the entire object as the default.

**Review by craicoverflow — CHANGES_REQUESTED** [review]
A couple of changes needed, but nice job 👍🏻

**Comment by alexal** [comment]
> A couple of changes needed, but nice job 👍🏻

Thanks for the feedback. I've made the changes, please take a look.

File: pkg/cmd/kafka/describe/describe.go [hunk 4]
```diff
@@ -125,6 +129,11 @@ func runDescribe(opts *Options) error {
 		}
 	}
 
+	if opts.bootstrapServer {
+		fmt.Fprintf(os.Stdout, "%v", *kafkaInstance.BootstrapServerHost)
```
Comment (craicoverflow) [inline pkg/cmd/kafka/describe/describe.go:133]:
```suggestion
		fmt.Println(os.Stdout, kafkaInstance.GetBootstrapServerHost())
```

^ Untested, but this is preferred. `IOStreams` is a an abstraction of stderr and stdout which we use instead.

Comment (craicoverflow) [reply pkg/cmd/kafka/describe/describe.go:133]: It is possible for the bootstrap server URL to be nil. We should handle this with a message.

Comment (alexal) [reply pkg/cmd/kafka/describe/describe.go]: Done, but we still have to use fmt.Fprintf function
`fmt.Fprintf(opts.IO.Out, kafkaInstance.GetBootstrapServerHost())`

**Review by wtrocki — APPROVED** [review]
Thank you so much for contribution.

File: pkg/cmd/kafka/describe/describe.go [hunk 5]
```diff
@@ -125,6 +127,11 @@ func runDescribe(opts *options) error {
 		}
 	}
 
+	if opts.bootstrapServer {
+		fmt.Fprintln(opts.IO.Out, kafkaInstance.GetBootstrapServerHost())
+		return nil
+	}
```
Comment (craicoverflow) [inline pkg/cmd/kafka/describe/describe.go:133]: I think we should be checking if the bootstrap_server_host exists, and if it does not print some info message letting the user know this.

Comment (wtrocki) [reply pkg/cmd/kafka/describe/describe.go:133]: Agree. We can extract this code into helper: 
https://github.com/redhat-developer/app-services-cli/blob/main/pkg/connection/keycloak_connection.go#L254-L267

Comment (alexal) [reply pkg/cmd/kafka/describe/describe.go]: I've added a message, please take a look.

File: pkg/localize/locales/en/cmd/kafka.en.toml [hunk 6]
```diff
@@ -402,6 +402,13 @@ one = 'Unique ID of the Kafka instance you want to view'
 description = 'Description for the --name flag'
 one = 'Name of the Kafka instance you want to view'
 
+[kafka.describe.flag.bootstrapserver]
```
Comment (wtrocki) [inline pkg/localize/locales/en/cmd/kafka.en.toml:405]: In long run we want to have ability to get some specific config of the services in unified fassion. 

It is good to have this in describe for now, but we should have similar capability for the status command

**Comment by wtrocki** [comment]
LGTM

**Comment by wtrocki** [comment]
Reverified. Good job!

File: pkg/localize/locales/en/cmd/kafka.en.toml [hunk 7]
```diff
@@ -406,6 +406,9 @@ one = 'Name of the Kafka instance you want to view'
 description = 'Description for the --bootstrap-server flag'
 one = 'If specified, only bootstrap server host of the Kafka instance will be displayed'
 
+[kafka.describe.bootstrapserver.not.available]
+one = 'Unable to get bootstrap server host name'
```
Comment (wtrocki) [inline pkg/localize/locales/en/cmd/kafka.en.toml:410]: get Kafka bootstrapServerHost

**Review by wtrocki — CHANGES_REQUESTED** [review]
Small message fix. Works good

File: pkg/localize/locales/en/cmd/kafka.en.toml (hunk 7 above)
Comment (alexal) [reply pkg/localize/locales/en/cmd/kafka.en.toml]: Looks better now?

File: pkg/localize/locales/en/cmd/kafka.en.toml [hunk 8]
```diff
@@ -402,6 +402,13 @@ one = 'Unique ID of the Kafka instance you want to view'
 description = 'Description for the --name flag'
 one = 'Name of the Kafka instance you want to view'
 
+[kafka.describe.flag.bootstrapserver]
+description = 'Description for the --bootstrap-server flag'
+one = 'If specified, only bootstrap server host of the Kafka instance will be displayed'
+
+[kafka.describe.bootstrapserver.not.available]
+one = 'Unable to get bootstrap server host name'
```
Comment (craicoverflow) [inline pkg/localize/locales/en/cmd/kafka.en.toml]:
```suggestion
one = 'Kafka instance "{{.Name}}" does not have a bootstrap server URL.'
```

usually we give more specific information. It is not that we are unable, it does not exist.

Comment (alexal) [reply pkg/localize/locales/en/cmd/kafka.en.toml]: @craicoverflow done, please take a look.

**Pull request merged**